### PR TITLE
fix(defaulttiming): issues fixed related to default timings

### PR DESCRIPTION
**Pull request by msohailhussain**
## Summary
Polling interval should be default when 0 or -ve value provided
Blocking timeout should be default when 0 or -ve value provided
Due call, should be made after releasing lock. 
The "why", or other context.

## Issues
- "THING-1234" or "Fixes #123"

### CODE REVIEW

File: OptimizelySDK/Config/HttpProjectConfigManager.cs [hunk 1]
```diff
@@ -122,14 +122,18 @@ protected override ProjectConfig Poll()
         
         public class Builder
         {
+            private const long MAX_MILLISECONDS_LIMIT = 4294967294;
```
Comment (mikeng13) [inline OptimizelySDK/Config/HttpProjectConfigManager.cs]: Cool, we need to make sure that we document these values for options in the README in your other PR

Comment (msohailhussain) [reply OptimizelySDK/Config/HttpProjectConfigManager.cs]: > Are these compat suite tests that we can include with this fix to ensure that other SDKs aren't similarly affected?

Sure, we can add FSC test cases.

**Comment by msohailhussain** [comment]
https://github.com/optimizely/csharp-sdk/pull/172#pullrequestreview-260333443
Sure, will add FSC test cases for these scenarios

File: OptimizelySDK/Config/HttpProjectConfigManager.cs (hunk 1 above)
Comment (msohailhussain) [reply OptimizelySDK/Config/HttpProjectConfigManager.cs]: Sure, will add this value in README.md

**Pull request merged**